experiment/simulation/js/layout.js

```diff
@@ -0,0 +1,144 @@
+import { deleteFA } from "./fa.js";
+import { deleteElement } from "./gate.js";
+import {
+  connectMux,
+  unbindEvent,
+  initALU,
+  refreshWorkingArea,
+} from "./main.js";
+import { simulateMux,deleteMux } from "./mux.js";
+"use strict";
+// Wires Colours
+export const wireColours = [
+  "#ff0000",
+  "#00ff00",
+  "#0000ff",
+  "#bf6be3",
+  "#ff00ff",
+  "#00ffff",
+  "#ff8000",
+  "#00ff80",
+  "#80ff00",
+  "#ff0080",
+  "#8080ff",
+  "#c0c0c0",
+];
+
+// Contextmenu
+const menu = document.querySelector(".menu");
+const menuOption = document.querySelector(".menu-option");
+let menuVisible = false;
+window.simulateMux = simulateMux;
+const toggleMenu = (command) => {
+  menu.style.display = command === "show" ? "block" : "none";
+  menuVisible = !menuVisible;
```

Comment:
menuVisible should always be computed from the menu style, best option for it is to be a function, otherwise there's always a risk of them going out of sync if the developer forgets to update them both at the same time.

Reply:
There was some bug coming when deriving from visibility property that's why as a workaround we stored it in a variable menu visible which indicates whether it's visible or not.